src/impl_aarch64__base_implementation.inl

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cpu_features_macros.h"
+
+#ifdef CPU_FEATURES_ARCH_AARCH64
+#if defined(CPU_FEATURES_OS_LINUX) || defined(CPU_FEATURES_OS_ANDROID) || \
+    defined(CPU_FEATURES_OS_MACOS)
```

Comment:
could there be an `CPU_FEATURES_OS_IPHONE` here?

Reply:
I don't have the hardware to test. I have added this macro. is it possible to test it at your end?

Reply:
I can test on an aarch64 raspberry pi running Ubuntu, but don't have an easy setup to test Android.

Reply:
I can test the same only on aarch64 rpi Ubuntu

Reply:
I have verified that it works on linux, for iphone `sys/sysctl.h` it should theoretically work: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/sysctl.3.html

Reply:
thank you @toor1245 @sbehnke